draft-ietf-anima-constrained-join-proxy.md

```diff
@@ -201,144 +232,217 @@ A Join Proxy can operate in two modes:
+belong to a single Plegdes UDP connection / DTLS/CoAP connection. The
+JPY Content field payload is the UDP payload of the packet for that UDP
+connection. Packets with different H belong to different Pledges UDP connections.
 
-The {{fig-stateless}} depicts the message flow diagram:
+In the stateless join proxy mode, both the Registrar and the Join Proxy use discoverable UDP join-ports. For the Join Proxy this may be a default CoAP port.
 
-~~~~
-+--------------+------------+---------------+-----------------------+
-|    Pledge    | Join Proxy |    Registrar  |        Message        |
-|     (P)      |     (J)    |      (R)      |Src_IP:port|Dst_IP:port|
-+--------------+------------+---------------+-----------+-----------+
-|      --ClientHello-->                     | IP_P:p_P  |IP_Jl:p_Jl |
-|                    --JPY[H(IP_P:p_P),-->  | IP_Jr:p_Jr|IP_R:p_Ra  |
-|                          C(ClientHello)]  |           |           |
-|                    <--JPY[H(IP_P:p_P),--  | IP_R:p_Ra |IP_Jr:p_Jr |
-|                         C(ServerHello)]   |           |           |
-|      <--ServerHello--                     | IP_Jl:p_Jl|IP_P:p_P   |
-|              :                            |           |           |
-|          [ DTLS messages ]                |     :     |    :      |
-|              :                            |     :     |    :      |
-|      --Finished-->                        | IP_P:p_P  |IP_Jr:p_Jr |
-|                    --JPY[H(IP_P:p_P),-->  | IP_Jl:p_Jl|IP_R:p_Ra  |
-|                          C(Finished)]     |           |           |
-|                    <--JPY[H(IP_P:p_P),--  | IP_R:p_Ra |IP_Jr:p_Jr |
-|                         C(Finished)]      |           |           |
-|      <--Finished--                        | IP_Jl:p_Jl|IP_P:p_P   |
-|              :                            |     :     |    :      |
-+-------------------------------------------+-----------+-----------+
-IP_P:p_P = Link-local IP address and port of the Pledge
-IP_R:p_Ra = Routable IP address and join-port of Registrar
-IP_Jl:p_Jl = Link-local IP address and join-port of Join Proxy
-IP_Jr:p_Jr = Routable IP address and port of Join Proxy
-
-JPY[H(),C()] = Join Proxy message with header H and content C
+Because there is only per-packet mapping of the IP / UDP and JPY header of
+received UDP packets according to {{fig-mapping-stateless}}, the stateless
+mode has no equivalent to {{fig-state-machinery}}. Therefore there is also
+no need for any of the requirements against this mapping state (as in
+{{stateful}}) for the stateless operation of the join proxy.
 
-~~~~
-{: #fig-stateless title='constrained stateless joining message flow.' align="left"}
+Unlike the stateful operation, ICMP error messages from the Registrar can not be
+mapped to the Pledge, because the ICMP error messages do not carry enough
+bytes of the original packets payload to include the JPY Header.
```

Comment:
Interesting point on ICMP / ICMPv6 errors. The last sentence (reason) isn't fully clear to me. Do you mean that the Registrar's ICMP error message doesn't contain the IPv6 header of the original Pledge's packet ? So the JP cannot send back an ICMP error message because it doesn't know how the original IPv6 header of the Pledge's packet looked like.   

Same issue could occur for a stateful JP however: it typically doesn't store full IPv6 headers but only some limited data like (address, port, timeout-variable) per Pledge?

The Registrar may be a 'normal' unconstrained IPv6 host so it could send out a complete ICMPv6 error message that includes the JPY header and all the other (DTLS) data.

Noteworthy is that 6lowpan mesh network nodes may only send the IPv6 header in the ICMPv6 error message, but not the full packet - to avoid high load on the mesh.  (The full data is in practice not used by the sender. The IPv6 header is enough to identify the sent message.) See for source code example of OpenThread mesh: https://github.com/openthread/openthread/blob/main/src/core/net/icmp6.cpp#L119

If the Registrar is located on the Border Router and thus a mesh node, it may also use this "limited" style of ICMPv6 errors with only IPv6 header info.  In that case the stateless JP can't find back its JPY header in the ICMPv6 error message from Registrar.